python/templates/macros/implementations.jinja2

```diff
@@ -19,7 +19,24 @@
 }
 
 Mutable{{ type }} {{ full_type }}::clone() const {
+{% if prefix %}
   return Mutable{{ type }}(podio::utils::MaybeSharedPtr(new {{ type }}Obj(*m_obj), podio::utils::MarkOwned));
+{% else %}
```

Comment:
Do I understand correctly here that if we `clone` a `Mutable` type, we are guaranteed to not have slicing and skip the rest?

Reply:
I think so, you can only get the pointers to the big vector when reading and then that's a non-mutable object which is now fixed so when you go to Mutable it should be fine. Previously I think it may have been possible by non-mutable (from reading) -> mutable (first clone) -> mutable (second clone) and the second clone would maybe be pointing to the big vector.